runtime/module.go

```diff
@@ -179,6 +179,67 @@ func ImportNativeModule(f *Frame, name string, members map[string]*Object) (*Obj
 	return prev, nil
 }
 
+// LoadMembers scans over all the members in module
+// and populates globals with them, taking __all__ into
+// account.
+func LoadMembers(f *Frame, module *Object) *BaseException {
+	allAttr, raised := GetAttr(f, module, NewStr("__all__"), nil)
+	if raised != nil && !raised.isInstance(AttributeErrorType) {
+		return raised
+	}
+	f.RestoreExc(nil, nil)
+
+	if raised == nil {
+		raised = loadMembersFromIterable(f, module, allAttr, nil)
+		if raised != nil {
+			return raised
+		}
+		return nil
+	}
+
+	// Fall back on __dict__
+	dictAttr, raised := GetAttr(f, module, NewStr("__dict__"), nil)
+	if raised != nil && !raised.isInstance(AttributeErrorType) {
+		return raised
+	}
+	raised = loadMembersFromIterable(f, module, dictAttr, func(key *Object) bool {
+		return strings.HasPrefix(toStrUnsafe(key).value, "_")
+	})
+	if raised != nil {
+		return raised
+	}
+	return nil
+}
+
+func loadMembersFromIterable(f *Frame, module, iterable *Object, filterF func(*Object) bool) *BaseException {
+	iter, raised := Iter(f, iterable)
+	if raised != nil {
+		return raised
+	}
+
+	memberName, raised := Next(f, iter)
+	for ; raised == nil; memberName, raised = Next(f, iter) {
+		member, raised := GetAttr(f, module, toStrUnsafe(memberName), nil)
```

Comment:
So I read the [source code](https://github.com/python/cpython/blob/9c1426de7521299f70eb5483e7e25d1c2a73dbbd/Python/ceval.c#L4880) and it seems to raise several errors:
- lines 48888 and 4893 seem to be unexpected errors, don't think we have to map these
- line 4895 if the module doesn't contain either one of __all__ and __dict__. It doesn't seem like it's possible in the Go code
- line 4909 if __all__ or __dict__ (whichever gets chosen) is not iterable. Throws IndexError
- line 4924 if the member name is not a string, or the module does not contain the member
- lines 4926 and 4928 throw errors if locals is malformed, the member name is already validated from above.

Looking over the exceptions, there are already a few covered ones, without changing the code:
- IndexError when trying to iterate over a non-iterable object
- AttributeError when the module does not have a requested member

So, what I should focus on is:
- TypeError is the member name is not actually a string
- SystemError if the Globals (locals) is malformed. I don't know how to replicate that in real-life though, so I couldn't test in the shell, I just went through the code.

WDYT?